packages/cadl-typescript/CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+```
+npm run test
+```
+
+Running the command above will do the following things:
+
+- Start TestServer
+- Build Cadl Typescript
+- Generate all scenarios in parallel (i.e. Dictionary, Extensible Enums, Models, Resiliency)
+- Run all the tests under test/integration
+- Stop TestServer
+
+**_Note_**: If your development environment is Windows, then run the command `npm run start-test-server:v2`(in a seperate window) before running `npm run test` and run the command `npm run stop-test-server` after. (In non windows machines, we could run the test-server in the background automatically. But, in Windows machines, it has to be done manually.)
+
+5. You can run the Smoke tests using the following command:
+
+```
+cd ../../packages/cadl-rlc-test
+npm run smoke-test
+```
+
+## How to add an integration test case
+
+Whenever you work on adding a feature/fixing a bug, this would probably be your first step. You create a test case and then run it through the generator, see the result, modify the generator, run it again and so on, until you get the desired output.
+
+1. Pick up a cadl as your test input in cadl-ranch. Below are some examples
+
+Let us say your test input will be called `testUserCase.json`.
+
+2. Now add an entry to the CadlRanchConfig to the file [`cadl-ranch-list.ts`](./test/commands/cadl-ranch-list.ts). In the file, add the following to the array.
+
+```
+  {
+    outputPath: "authentication/apiKey",
+    inputPath: "authentication/api-key"
+  },
+```
+
+3. Now, You can generate the RLC for your test case with the following command: (Initially, during your development, you do not want to run all the cases during every step of your development, you can comment out other test cases. But, once your code changes are complete for your case, then you need to run the entire suite to ensure that your changes did not cause any unwanted changes.)
+
+```
+npm run generate-cadl-only
+```
+
+4. Once you are satisfied with the generated code, you can add a spec file such as `testUserCaseRest.spec.ts` file [here](./test/integration). You can find several examples in the same place.
+
+## How to debug
+
+### `generate-cadl-only` step
+
+If you would like to debug the `generate-cadl-only` step for our test input, use the following command:
+
+```
+npm run generate-cadl-only -- -b -i testUserCase --debug
+```
+
+Now, the code will wait for the debugger to be attached. Open the repository in VS Code -> Select `Run and Debug` section -> Click `Attach`.
+
+### Spec file
+
+If you would like to debug the `testUserCase.spec.ts` file (after the SDK is generated), Open the repository in VS Code -> Open the `testUserCase.spec.ts` file -> Select `Run and Debug` section -> Click `IntegrationTests - Current File`.
+
+### How to debug an unit test case
+
+- In VS Code, We have created a Debugging profile for UnitTests to start debugging:
+
+  1. Go to the debugger tab
+  2. Select the "Unit Test" Profile
```

Comment:
Do you mean we could debug cadl UT by config `Cadl Unit or Integration Test`?

Reply:
updated